test/parallel/test-stream-writable-ended-state.js

```diff
@@ -0,0 +1,20 @@
+'use strict';
+
+require('../common');
+
+const assert = require('assert');
+const stream = require('stream');
+
+const writable = new stream.Writable();
+
+writable._write = (chunk, encoding, cb) => {
+  cb();
+};
+
+assert.strictEqual(writable._writableState.ended, false);
+
+writable.end('testing ended state', () => {
```

Comment:
The callback here should be wrapped in a `common.mustCall()` in order to verify that it actually does get called.